# Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
#

Since libXfont is used by programs like mkfontdir & installalias that run
during Live Upgrade from old Solaris releases to new ones, we need to
minimize dependencies on library versions found only on new releases.

This patch uses macros for htons() & ntohs() to avoid creating a dependency
on the Nevada version of libc (SUNW_1.23) which has added those functions to
libc (they were previously in libnsl on older releases).

Since those are already no-op macros on SPARC, this is not needed there.

--- src/fc/fstrans.c	Mon May 12 14:31:00 2008
+++ src/fc/fstrans.c	Fri Sep 12 19:22:31 2008
@@ -19,7 +19,13 @@
  * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  * PERFORMANCE OF THIS SOFTWARE.
  */
  
+#ifndef __sparc
+/* inline htons() & ntohs() to avoid dependency on libc version SUNW_1.23 */
+#define htons(x) BSWAP_16(x)
+#define ntohs(x) BSWAP_16(x)
+#endif
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
